internal/metrics/datanode.go

```diff
@@ -0,0 +1,240 @@
+			"num_flow_graphs",
+			nodeIDLabelName,
+		})
+
+	DataNodeConsumeMsgRowsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "message_size_count",
+			Help:      "Messages size count consumed from msgStream in DataNode.",
+		}, []string{
+			"message_size_count",
+			nodeIDLabelName,
+		})
+
+	DataNodeFlushedSize = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "flushed_size",
+			Help:      "Data size flushed to storage in DataNode.",
+		}, []string{
+			"flushed_size",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumDmlChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_dml_channels",
+			Help:      "Number of dmlChannels per collection in DataNode.",
+		}, []string{
+			"num_dml_channels",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumDeltaChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_delta_channels",
+			Help:      "Number of deltaChannels per collection in DataNode.",
+		}, []string{
+			"num_delta_channels",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumConsumers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_consumers",
+			Help:      "Number of consumers per collection in DataNode.",
+		}, []string{
+			"num_consumers",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumProducers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_producers",
+			Help:      "Number of producers per collection in DataNode.",
+		}, []string{
+			"num_producers",
+			nodeIDLabelName,
+		})
+
+	DataNodeTimeSync = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "time_sync_per_channel",
+			Help:      "Synchronized timestamps per channel in DataNode.",
+		}, []string{
+			"time_sync_per_channel",
+			nodeIDLabelName,
+		})
+
+	DataNodeSegmentRowsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "seg_rows_count",
+			Help:      "Rows count of segments which sent to DataCoord from DataNode.",
+		}, []string{
+			"seg_rows_count",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumUnflushedSegments = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_unflushed_segments",
+			Help:      "Number of unflushed segments in DataNode.",
+		}, []string{
+			"num_unflushed_segments",
+			nodeIDLabelName,
+		})
+
+	DataNodeFlushDuration = prometheus.NewHistogramVec(
```

Comment:
his refers to the time interval between the datanode receiving the flush request and the completion of the flush operation.

Reply:
arguably